ui/raidboss/data/05-shb/alliance/the_puppets_bunker.js

```diff
@@ -1,11 +1,547 @@
+      },
+      run: (data) => delete data.busterTargets,
+    },
+    {
+      id: 'Puppet Superior Shield Protocol',
+      netRegex: NetRegexes.startsUsing({ id: '4FA[678]', capture: false }),
+      run: (data) => data.phase = 'superior',
+    },
+    {
+      id: 'Puppet Superior Missile Command',
+      netRegex: NetRegexes.startsUsing({ id: '4FBD', capture: false }),
+      condition: Conditions.caresAboutAOE(),
+      suppressSeconds: 5,
+      response: Responses.aoe(),
+    },
+    {
+      // This is for Maneuver: Incendiary Bombing and Maneuver: Area Bombardment.
+      id: 'Puppet Superior Incendiary Bombing',
+      netRegex: NetRegexes.headMarker({ id: '0017' }),
+      condition: (data, matches) => data.me === matches.target && data.phase === 'superior',
+      alertText: {
+        en: 'Fire Puddle on YOU',
+      },
+    },
+    {
+      id: 'Puppet Superior High-Powered Laser',
+      // Note: no 1B marker for this???
+      netRegex: NetRegexes.startsUsing({ id: '4FB4', capture: false }),
+      suppressSeconds: 5,
+      alertText: {
+        en: 'Line Stack (Point Outside)',
+      },
+    },
+    {
+      // TODO: we could theoretically get combatants here when sharp turn starts casting.
+      // Sharp Turn ability ids differentiate left/right.  We could use effect ids to know group.
+      // But how would you communicate where to stand? "behind and slightly/very right"?
+      id: 'Puppet Superior Sharp Turn',
+      netRegex: NetRegexes.startsUsing({ id: '4FAB', capture: false }),
+      delaySeconds: 3,
+      suppressSeconds: 5,
+      alertText: {
+        en: 'Dodge Sword Charges',
+      },
+    },
+    {
+      id: 'Puppet Superior Precision Guided Missile You',
+      netRegex: NetRegexes.startsUsing({ id: '4FC5' }),
+      condition: Conditions.targetIsYou(),
+      response: Responses.tankBuster('alert'),
+    },
+    {
+      id: 'Puppet Superior Precision Guided Missile Collect',
+      netRegex: NetRegexes.startsUsing({ id: '4FC5' }),
+      run: function(data, matches) {
+        data.busterTargets = data.busterTargets || [];
+        data.busterTargets.push(matches.target);
+      },
+    },
+    {
+      id: 'Puppet Superior Precision Guided Missile Not You',
+      netRegex: NetRegexes.startsUsing({ id: '4FC5', capture: false }),
+      delaySeconds: 0.5,
+      suppressSeconds: 5,
+      infoText: function(data) {
+        if (!data.busterTargets)
+          return;
+        if (data.busterTargets.includes(data.me))
+          return;
+
+        if (data.role === 'healer') {
+          return {
+            en: 'Tank Buster',
+            de: 'Tank buster',
+            fr: 'Tank buster',
+            ja: 'タンクバスター',
+            cn: '坦克死刑',
+            ko: '탱버',
+          };
+        }
+        return {
+          en: 'Avoid tank buster',
+        };
+      },
+      run: (data) => delete data.busterTargets,
+    },
+    {
+      id: 'Puppet Superior Sliding Swipe First',
```

Comment:
This is like take 3 on this code.  It still hasn't worked yet.

Reply:
This looks like a reasonable way of handling it. I think even as annoying as copying text from `responses` is, it's the right thing to do for now.

Reply:
Ok, take 4 seems correct.  (oops pop vs shift)